reactor-core/src/main/java/reactor/core/publisher/FluxIndex.java

```diff
@@ -274,5 +274,15 @@ public I apply(Long i, T t) {
 			}
 			return typedIndex;
 		}
+
+		static <T, I> BiFunction<? super Long, ? super T, ? extends I> create(
+				BiFunction<? super Long, ? super T, ? extends I> indexMapper) {
+			if (indexMapper == Flux.TUPLE2_BIFUNCTION) {
+				// TUPLE2_BIFUNCTION (Tuples::of) never returns null.
+				// Also helps FluxIndexFuseable to detect the default index mapper.
```

Comment:
@dittos can you expand on that part? doesn't seem to ever be checked apart from here

Reply:
maybe add a test that instantiating with that default mapper doesn't override the internal `indexMapper` field?

Reply:
`requestFusion` checks for it.

https://github.com/reactor/reactor-core/blob/0328059f370b03c9212ccb53b92197eeb2ed430b/reactor-core/src/main/java/reactor/core/publisher/FluxIndexFuseable.java#L195

There is a test for the default mapper + thread barrier case, so I think the comment is enough (instead of adding test). Maybe need to word it better.

https://github.com/reactor/reactor-core/blob/0328059f370b03c9212ccb53b92197eeb2ed430b/reactor-core/src/test/java/reactor/core/publisher/FluxIndexedFuseableTest.java#L237-L245

Reply:
ah right it was a pre-existing test :+1: